### PR TITLE
chore: bump typescript-eslint to v8

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
👋 Hi! Coming over from https://github.com/typescript-eslint/typescript-eslint/issues/9501: we're working on typescript-eslint v8 and would like to try the beta out on plugins such as `eslint-plugin-jsonc`.

I'm sending this draft PR as a reference to see what issues we might give you in upgrading and to try to be helpful. If this is annoying noise to you, please forgive me, I'll withdraw the PR. But I hope this is useful on your end - and please let me know if you have any feedback! ❤️

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -46,7 +46,7 @@ export default [
       ecmaVersion: "latest",
       sourceType: "script",
       parserOptions: {
-        project: true,
+        projectService: true,
```
Comment (JoshuaKGoldberg) [inline eslint.config.mjs:49]: The same issue as https://github.com/ota-meshi/eslint-plugin-yml/pull/346#discussion_r1667218685! ❗ 

```plaintext

/Users/josh/repos/eslint-plugin-jsonc/docs/.vitepress/theme/components/components/EslintPluginEditor.vue
  0:0  error  Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): docs/.vitepress/theme/components/components/EslintPluginEditor.vue

/Users/josh/repos/eslint-plugin-jsonc/docs/.vitepress/theme/components/components/PgEditor.vue
  0:0  error  Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): docs/.vitepress/theme/components/components/PgEditor.vue

/Users/josh/repos/eslint-plugin-jsonc/docs/.vitepress/theme/components/components/RulesSettings.vue
  0:0  error  Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): docs/.vitepress/theme/components/components/RulesSettings.vue

/Users/josh/repos/eslint-plugin-jsonc/docs/.vitepress/theme/components/components/SnsBar.vue
  0:0  error  Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): docs/.vitepress/theme/components/components/SnsBar.vue

/Users/josh/repos/eslint-plugin-jsonc/docs/.vitepress/theme/components/eslint-code-block.vue
  0:0  error  Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): docs/.vitepress/theme/components/eslint-code-block.vue

/Users/josh/repos/eslint-plugin-jsonc/docs/.vitepress/theme/components/playground-block.vue
  0:0  error  Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): docs/.vitepress/theme/components/playground-block.vue
```

Comment (JoshuaKGoldberg) [reply eslint.config.mjs:49]: As with that issue, this should be fixed as https://github.com/typescript-eslint/typescript-eslint/pull/9580.

**Comment by JoshuaKGoldberg** [comment]
Now that [typescript-eslint v8 released is](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8), updating to `8.0.0` and un-drafting.

File: lib/utils/get-auto-jsonc-rules-config/index.ts
Comment (JoshuaKGoldberg) [inline lib/utils/get-auto-jsonc-rules-config/index.ts:1]: > `Replace `rule.split("/").pop()·??·rule` with `(rule.split("/").pop()·??·rule)` eslint(prettier/prettier)`

Is this a pre-existing failure? Should I do something in this PR?

**Review by ota-meshi — APPROVED** [review]
Thank you so much!

**Pull request merged**